src/Compression/CompressedWriteBuffer.cpp

```diff
@@ -50,9 +50,25 @@ CompressedWriteBuffer::CompressedWriteBuffer(
 
 CompressedWriteBuffer::~CompressedWriteBuffer()
 {
-    /// FIXME move final flush into the caller
-    MemoryTracker::LockExceptionInThread lock;
-    next();
+    if (hasPendingData())
+    {
+
+#ifndef NDEBUG
+        /// No exception, buffer was not flushed. Consider it as a logical error.
+        if (std::uncaught_exception() == 0)
+            std::terminate();
+#endif
+        /// Try our best to flush buffer.
+        MemoryTracker::LockExceptionInThread lock;
+        try
+        {
+            next();
+        }
+        catch (...)
+        {
+            tryLogCurrentException("CompressedWriteBuffer::~CompressedWriteBuffer");
```

Comment:
But is it OK to hide write errors (especially when debug build shows that every peace of code flushes the data appropriately)? I would vote for std::terminate instead of message in the [error] log